src/screens/Technology/components/Details/index.tsx

```diff
@@ -121,6 +128,119 @@ export const Characteristics = () => {
   );
 };
 
+interface MarkerProps {
+  type: string
+  [key: string]: any
+}
+
+export const Geo = (): JSX.Element => {
+  const technology = useTechnology();
+  const types: string[] = ['who_develop', 'where_can_be_applied', 'where_is_already_implemented'];
+  const region = {
+    latitude: -6.780127,
+    longitude: -36.702823,
+  };
+  const zoom = -1500;
+  const mapRef = useRef(null);
+
+  const [markers, setMarkers] = useState([]);
+  const [cachedMarkers, setCachedMarkers] = useState([]);
+  const [markerFilters, setMarkerFilters] = useState<string[]>(types);
+
+  const formatMarkers = (term: any) => {
+    const colors: string[] = ['blue', 'yellow', 'red'];
+    const marker: MarkerProps = { type: term.term };
+
+    term.metas.forEach(({ meta_key, meta_value }: any) => {
+      marker[meta_key] = meta_value;
+    });
+
+    return {
+      ...marker,
+      color: colors[types.indexOf(marker.type)],
+    };
+  };
+
+  useEffect(() => {
+    const getTerms = async () => {
+      const terms = await getTechnologyTerms(technology.id);
+      const formattedMarkers = terms.filter(({ term }: any) => markerFilters.includes(term)).map(formatMarkers);
+      setMarkers(formattedMarkers);
+      setCachedMarkers(formattedMarkers);
+    };
+
+    getTerms();
+
+    // Animate zoom MAP
+    mapRef.current.animateCamera({
+      center: region,
+      zoom,
+      altitude: zoomToAltitude(zoom),
+    }, 400);
+  }, []);
+
+  useEffect(() => {
+    setMarkers(cachedMarkers.filter(({ type }: any) => markerFilters.includes(type)));
+  }, [markerFilters]);
+
+  return (
+    <>
+      <S.MapWrapper>
+        <MapView
+          ref={mapRef}
+          loadingEnabled
+          zoomControlEnabled
+          initialRegion={{
+            ...region,
+            latitudeDelta: 0.0922,
+            longitudeDelta: 0.0421,
+          }}
+          style={{
+            height: 300,
+          }}
+        >
+          {markers.length > 0 && markers.map((marker) => (
+            <Marker
```

Comment:
You need to define a `key` attribute for this component because it's inside of a loop :thinking:

Reply:
@mateus4k nesse caso o component <Marker> não precisa desse atributo `key` mesmo que estando dentro de um loop

Reply:
Pq? 👀